tensorboard/webapp/runs/views/runs_table/sorting_utils.ts

```diff
@@ -0,0 +1,131 @@
+/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {
+  SortingInfo,
+  SortingOrder,
+  TableData,
+} from '../../../widgets/data_table/types';
+import {ExperimentAlias} from '../../../experiments/types';
+
+enum UndefinedStrategy {
+  BEFORE,
+  AFTER,
+}
+
+interface SortOptions {
+  insertUndefined: UndefinedStrategy;
+}
+
+const POTENTIALLY_NUMERIC = new Set(['string', 'number']);
```

Comment:
This name confused me. I understood it once I saw it used so I think it is probably ok. I would prefer POTENTIALLY_NUMERIC_TYPES.